platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/tipset/TipsetUtils.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.platform.event.tipset;
+
+import com.swirlds.platform.event.EventDescriptor;
+import com.swirlds.platform.event.GossipEvent;
+import com.swirlds.platform.internal.EventImpl;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Misc tipset utilities.
+ */
+public final class TipsetUtils {
+
+    private TipsetUtils() {}
+
+    /**
+     * Build a descriptor from an EventImpl.
+     *
+     * @param event the event
+     * @return the descriptor
+     */
+    public static EventDescriptor buildDescriptor(@NonNull final EventImpl event) {
```

Comment:
Would it make more sense to put this method in either the `EventDescriptor` or `EventImpl` class? It doesn't seem specific to tipset

Reply:
The reason why I'm keeping it isolated is because I don't like the current event descriptor API, and want to improve it. Ideally, both would just have a method `getDescriptor()`, but currently it's a lot more complex than that. These methods are just temporary workarounds that keep the important parts of the tipset algorithm clean.